crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -579,22 +579,23 @@ def upgrade_compute_nodes(virt)
           next if n.upgraded?
           node_api = Api::Node.new n.name
           node_api.save_node_state("compute")
+          hostname = n[:hostname] || n.name
```

Comment:
I think a small comment why ```hostname``` instead of ```name``` is preferrable here (or on the commit message), would help.